res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -1572,60 +1810,25 @@ PioneerDDJSX.playLed = function(value, group, control) {
     PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.shiftPlay, value);
 };
 
-PioneerDDJSX.playFromCuePoint = false;
-
-PioneerDDJSX.playIllumination = function(value, group, control) {
+PioneerDDJSX.wheelLeds = function(value, group, control) {
     var deck = PioneerDDJSX.channelGroups[group],
-        cue = PioneerDDJSX.cuePressed[deck],
-        trackLoaded = 0;
+        duration = engine.getValue(group, "duration"),
+        remainingTime = duration - (value * duration),
+        speed = 0x28,
+        wheelPos = 0x01;
 
-    trackLoaded = engine.getValue(group, "track_loaded");
-    if (!trackLoaded) {
-        return;
+    if (value >= 0) {
+        wheelPos = 0x01 + ((speed * (value * duration)) % 0x48);
```

Comment:
`speed` and `0x48` are no magic numbers. `speed` represents the turntable rotation (33 1/3 RPM), it is calculated on paper before. I will change the code so you will see the calculation according scratch RPM parameter. `0x48` is the maximum hex value/midi message value representing all LEDs are on (one complete wheel turn). I will add intermediate variables to make the calculation visible and the code clearer.